src/game/enemies/kyacchaa.c

```diff
@@ -0,0 +1,468 @@
+
+    s->oamFlags = SPRITE_OAM_ORDER(19);
+    s->animCursor = 0;
+    s->qAnimDelay = 0;
+    s->animSpeed = SPRITE_ANIM_SPEED(1.0);
+    s->palId = 0;
+    s->frameFlags = SPRITE_FLAG(PRIORITY, 1);
+
+    s->hitboxes[0].index = HITBOX_STATE_INACTIVE;
+
+    UpdateSpriteAnimation(s);
+}
+
+void sub_8065A8C(void)
+{
+    Kyacchaa *enemy = TASK_DATA(gCurTask);
+    bool32 checkResult = FALSE;
+
+    if ((gStageData.unk4 != 1) && (gStageData.unk4 != 2) && (gStageData.unk4 != 4)) {
+        checkResult = sub_8065C48(enemy);
+    }
+
+    sub_8065CE0(enemy);
+
+    if (sub_8065F5C(enemy) == TRUE) {
+        TaskDestroy(gCurTask);
+        return;
+    }
+
+    if (checkResult == TRUE) {
+        Sprite *s2 = &enemy->s2;
+
+        s2->anim = gUnknown_080D2198[1].anim;
+        s2->variant = gUnknown_080D2198[1].variant;
+        s2->prevVariant = -1;
+
+        UpdateSpriteAnimation(s2);
+
+        gCurTask->main = Task_8065B0C;
+    }
+}
+
+void Task_8065B0C(void)
+{
+    Kyacchaa *enemy = TASK_DATA(gCurTask);
+    AnimCmdResult acmdRes = sub_8065CE0(enemy);
+
+    if (sub_8065F5C(enemy) == TRUE) {
+        TaskDestroy(gCurTask);
+        return;
+    }
+
+    if ((gStageData.unk4 != 1) && (gStageData.unk4 != 2) && (gStageData.unk4 != 4)) {
+        if (acmdRes == ACMD_RESULT__ENDED) {
+            enemy->unkC++;
+
+            if (enemy->unkC > 60) {
+                Sprite *s = &enemy->s2;
+
+                s->anim = gUnknown_080D2198[2].anim;
+                s->variant = gUnknown_080D2198[2].variant;
+                s->prevVariant = -1;
+
+                UpdateSpriteAnimation(s);
+                enemy->unkC = 0;
+
+                gCurTask->main = sub_8065E48;
+            }
+        }
+    }
+}
+
+bool32 sub_8065B90(Kyacchaa *enemy)
+{
+    s32 worldX, worldY;
+    u8 i;
+    Player *p;
+
+    worldX = I(enemy->qPos.x);
+    worldY = I(enemy->qPos.y);
+    worldX = TO_WORLD_POS_RAW(worldX, enemy->region[0]);
+    worldY = TO_WORLD_POS_RAW(worldY, enemy->region[1]);
+
+    for (i = 0; i < 2; i++) {
+        p = sub_805CD20(i);
+        if (!p) {
+            break;
+        }
+
+        // First check: player to the right of enemy within range
+        if (p->qWorldX - Q(worldX) < Q(40) && p->qWorldX >= Q(worldX)) {
+            // Vertical check
+            if (p->qWorldY - Q(worldY) < Q(80) && p->qWorldY >= Q(worldY)) {
+                // Store relative position in enemy's unk10/unk14 fields
+                goto lbl;
+            }
+        }
+
+        // Second check: player to the left of enemy within range
+        if (Q(worldX) - p->qWorldX < Q(40) && Q(worldX) >= p->qWorldX) {
+            // Vertical check
+            if (p->qWorldY - Q(worldY) < Q(80) && p->qWorldY >= Q(worldY)) {
+            lbl:
+                // Store relative position in enemy's unk10/unk14 fields
+                enemy->qUnk10.x = p->qWorldX - (enemy->region[0] << 16);
+                enemy->qUnk10.y = p->qWorldY - (enemy->region[1] << 16);
```

Comment:
I expected that I could use `TO_WORLD_POS`, but since it's shifted by 16 I could not use...
Am I missing something?